Complete isfrozen function term entry (#3317)
Codecademy/docs: content/luau/concepts/tables/terms/isfrozen/isfrozen.md

---
Title: 'isfrozen()'
Description: 'Returns a boolean based on whether the table is in read-only mode or not.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Booleans'
  - 'Tables'
CatalogContent:
  - 'paths/computer-science'
---

The **`isfrozen()`** function in the Luau programming language returns a boolean based on whether the table is in read-only mode or not. When a table is in read-only mode, it means that the contents can't be modified, as a measure of data protection.

## Syntax

```pseudo
isfrozen(table);
```

> The 'isfrozen()' function returns a boolean value, true if the table is in read-only mode and false if it's not.

Reviewer: ```suggestion
The `isfrozen()` function returns a boolean value, `true` if the table is in read-only mode and `false` if it's not.
```

## Example

> ```luau

Reviewer: ```suggestion
```lua
```

Reviewer: I don't believe luau is supported by markdown code blocks, but Lua can be used to show the syntax highlighting

-- Create a table
local myTable = {1, 2, 3}

-- Check if the table is initially frozen (read-only)
local frozenStatus = table.isfrozen(myTable)

-- Output the result
if frozenStatus then
    print("The table is frozen (read-only).")
else
    print("The table is not frozen (read/write).")
end
```

> In this example, a table called `myTable` is created. Then, the `isfrozen()` function is used to check if it's in read-only mode. Depending on the result, a message is printed indicating whether the table is frozen (read-only) or not.

Reviewer: ```suggestion
In this example, a table called `myTable` is created. Then, the `isfrozen()` function is used to check if it's in read-only mode.

Reviewer: This is generally set before the example. And an output ```shell block is used to highlight the result

> 

Reviewer 2: ```suggestion
```
Remove blank line at start of entry.

---

> Title: '.isfrozen'

Reviewer 2: ```suggestion
Title: 'isfrozen()'
```

```shell
---
Title: '.isfrozen'
Description: 'Returns a boolean based on whether the table is in read-only mode or not.'
Subjects:
  - 'luau .isfrozen function'
Tags:
  - 'Boolean'
  - 'Table'
CatalogContent:
  - 'paths/computer-science'
---
```

## Introduction

The **isfrozen** function in the Luau programming language returms a boolean based on whether the table is in read-only mode or not. When a table is in read-only mode, it means that the contents can't be modified, as a measure of data protection.

## Syntax

```luau
isfrozen(table);
```

The 'isfrozen()' function returns a boolean value, true if the table is in read-only mode and false if it's not.

> ## Codebyte Example

Reviewer 2: ```suggestion
## Example
```
This isn't a codebyte. Codebytes are special code blocks that the reader can execute. It's only supported for some languages, and Luau isn't one of them.

```luau
-- Create a table
local myTable = {1, 2, 3}

-- Check if the table is initially frozen (read-only)

> local frozenStatus = myTable:IsFrozen()

Reviewer 2: ```suggestion
local frozenStatus = table.isfrozen(myTable)
```

-- Output the result
if frozenStatus then
    print("The table is frozen (read-only).")
else
    print("The table is not frozen (read/write).")
end
```

> In this example, we first create a table called myTable. Then, we use the isfrozen() function to check if it's in read-only mode. Depending on the result, we print a message indicating whether the table is frozen (read-only) or not.

Reviewer 2: ```suggestion
In this example, a table called `myTable` is created. Then, the `isfrozen()` function is used to check if it's in read-only mode. Depending on the result, a message is printed indicating whether the table is frozen (read-only) or not.
```
After these edits, let's move this before the example code block.
